code/controllers/stonedmc/subsystem/direction.dm

```diff
@@ -37,26 +39,40 @@ SUBSYSTEM_DEF(direction)
 	if(!resumed)
 		currentrun = deepCopyList(processing_mobs)
 
-	for(var/L in currentrun)
-		var/mob/living/carbon/human/H
-		if(iscarbon(leader_mapping[L]))
-			var/mob/living/carbon/C = leader_mapping[L]
-			while(currentrun[L].len)
-				H = currentrun[L][currentrun[L].len]
-				currentrun[L].len--
-				if(!H)
-					processing_mobs[L].Remove(H)
-					continue
-				H.update_leader_tracking(C)
-				if(MC_TICK_CHECK)
-					return
-		else
-			while(currentrun[L].len)
-				H = currentrun[L][currentrun[L].len]
-				currentrun[L].len--
-				H.clear_leader_tracking()
-				if(MC_TICK_CHECK)
-					return	
+	for(var/squad_id in currentrun)
+		var/mob/living/L
+		var/mob/living/SL = leader_mapping[squad_id]
+
+		if(QDELETED(SL) || !isliving(SL))
+			if (!clear_run(squad_id))
+				return
+
+		while(currentrun[squad_id].len)
+			L = currentrun[squad_id][currentrun[squad_id].len]
+			currentrun[squad_id].len--
+			if(QDELETED(L))

```

Comment:
Same here. If this is happening, somewhere else did an oof. Shouldn't be checking each time.